test/e2e/storage/testsuites/snapshotsecrets.go

```diff
@@ -0,0 +1,253 @@
+
+			pvc = volumeResource.Pvc
+			ginkgo.By("starting a pod to use the claim")
+
+			err = e2epv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, cs, pvc.Namespace, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("checking the claim")
+			// Get new copy of the claim
+			pvc, err = cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			// Get the bound PV
+			ginkgo.By("checking the PV")
+			_, err = cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+		}
+
+		cleanup := func() {
+			// Don't register an AfterEach then a cleanup step because the order
+			// of execution will do the AfterEach first then the cleanup step.
+			// Also AfterEach cleanup registration is not fine grained enough
+			// Adding to the cleanup steps allows you to register cleanup only when it is needed
+			// Ideally we could replace this with https://golang.org/pkg/testing/#T.Cleanup
+
+			// Depending on how far the test executed, cleanup accordingly
+			// Execute in reverse order, similar to defer stack
+			for i := len(cleanupSteps) - 1; i >= 0; i-- {
+				err := tryFunc(cleanupSteps[i])
+				framework.ExpectNoError(err, "while running cleanup steps")
+			}
+
+		}
+		ginkgo.BeforeEach(func() {
+			init()
+		})
+		ginkgo.AfterEach(func() {
+			cleanup()
+		})
+
+		ginkgo.Context("", func() {
+			var (
+				vs *unstructured.Unstructured
+				// vscontent *unstructured.Unstructured
+				// vsc    *unstructured.Unstructured
+				secret *v1.Secret
+			)
+
+			ginkgo.BeforeEach(func() {
+				var sr *SnapshotResource
+				cleanupSteps = append(cleanupSteps, func() {
+					framework.ExpectNoError(sr.CleanupResource())
+				})
+				// create a secret
+				secret = &v1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: f.Namespace.Name,
+						Name:      "snapshot-secret",
+					},
+					Data: map[string][]byte{
+						"secret-data": []byte("secret-value-1"),
+					},
+				}
+
+				if secret, err = cs.CoreV1().Secrets(f.Namespace.Name).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+					framework.Failf("unable to create test secret %s: %v", secret.Name, err)
+				}
+
+				// Create snapshot using secret
+				parameters := map[string]string{
+					"csi.storage.k8s.io/snapshotter-secret-name":      "snapshot-secret",
+					"csi.storage.k8s.io/snapshotter-secret-namespace": f.Namespace.Name,
+				}
+				sr = CreateSnapshotResource(sDriver, config, pattern, pvc.GetName(), pvc.GetNamespace(), parameters)
+				vs = sr.Vs
+			})
+			ginkgo.It("Use secrets field, create/list and delete snapshots", func() {
+
+				ginkgo.By("list the snapshot")
+				// TODO: Check if list snapshot is supported. Log and continue if not supported.
```

Comment:
Can you add this test?

Reply:
to check if list snapshot is supported, I will have to do a gRPC call to check if the capability: `csi.ControllerServiceCapability_RPC_LIST_SNAPSHOTS` is supported. Currently there is not test case that calls the makes the gRPC calls. I think I might have to generate the go code for csi proto file. correct?

Reply:
I see.  No, don't check if ListSnapshots is supported.  It is optional.  Snapshot operations will not fail if ListSnapshots is not supported.